tests/test_utils.py

```diff
@@ -66,17 +66,41 @@ def test_convert_to_float():
 
 def test_parse_json_with_comments():
```

Comment:
One more thing I was thinking: This test is pretty much the same test with slight modifications. Could the test simply be parameterized or could there be one test that uses a helper function instead? Even just a list of values with a for loop (or two for loops) might be good, e.g.

```python
for json_string in [json_string_inline, json_string_newline]:
    for http_or_https in ["http", "https"]:
        json_string = ....format(http_or_https)
        [do the rest of the stuff] 
```

Reply:
Good idea!

Reply:
@mulhod @desilinguist I thought that different functionalities or test cases must be tested in separate functions? Is that not the case? Otherwise, these test functions can be modified to test everything in one function, with two for loops.

Reply:
I don't think we need to have separate tests. If we use a test generator we can kind of get the best of both worlds, perhaps.

Reply:
Yeah, using a test generator is what I was thinking. That way you get the code reduction but also each test being logically separate.

Reply:
https://nose.readthedocs.io/en/latest/writing_tests.html#test-generators

Check this out.